### PR TITLE
dao-service done

### PR DESCRIPTION
**Pull request by not4und**

### CODE REVIEW

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            statement.setString(1, driver.getName());
+            statement.setString(2, driver.getLicenseNumber());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                driver.setId(resultSet.getObject(1, Long.class));
+            }
+            return driver;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't cteate driver: " + driver, e);
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:35]:
```suggestion
            throw new DataProcessingException("Couldn't create driver: " + driver, e);
```

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        }
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        String query = "SELECT * "
+                     + "FROM drivers "
+                     + "WHERE id = ? AND is_deleted = false;";
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:43]:
```suggestion
        String query = "SELECT * FROM drivers "
                     + "WHERE id = ? AND is_deleted = FALSE;";
```

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get driver by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Driver> getAll() {
+        String query = "SELECT * "
+                     + "FROM drivers "
+                     + "WHERE is_deleted = false;";
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:62]:
```suggestion
        String query = "SELECT * FROM drivers "
                     + "WHERE is_deleted = FALSE;";
```

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            throw new DataProcessingException("Couldn't get a list of drivers "
+                    + "from drivers table.", e);
+        }
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        String query = "UPDATE drivers "
+                     + "SET name = ?, license_number = ? "
+                     + "WHERE id = ? AND is_deleted = false;";
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:81]:
```suggestion
        String query = "UPDATE drivers SET name = ?, license_number = ? "
                     + "WHERE id = ? AND is_deleted = FALSE;";
```

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update a driver: " + driver, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE drivers "
+                     + "SET is_deleted = true "
+                     + "WHERE id = ?;";
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:98]:
```suggestion
        String query = "UPDATE drivers SET is_deleted = TRUE "
                     + "WHERE id = ?;";
```

File: src/main/resources/init_db.sql [hunk 6]
```diff
@@ -2,8 +2,16 @@ CREATE SCHEMA IF NOT EXISTS `taxi_service` DEFAULT CHARACTER SET utf8;
...
+        `is_deleted` TINYINT NOT NULL DEFAULT 0,
+        PRIMARY KEY (`id`)
+);
+CREATE TABLE `drivers` (
+        `id` BIGINT(11) NOT NULL AUTO_INCREMENT,
+        `name` VARCHAR(225) NOT NULL,
+        `license_number` VARCHAR(225) NOT NULL,
+        `is_deleted` TINYINT NOT NULL DEFAULT 0,
+         PRIMARY KEY (`id`)
+);
```
Comment (kozhukhovsky) [inline src/main/resources/init_db.sql:17]: add empty line

File: src/main/java/mate/jdbc/service/impl/DriverServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    private DriverDao dao;
+
+    @Override
+    public Driver create(Driver driver) {
+        return dao.create(driver);
+    }
+
+    @Override
+    public Driver get(Long id) {
+        return dao.get(id).get();
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/service/impl/DriverServiceImpl.java:22]: What happens, if optional is empty?
```suggestion
        return dao.get(id).orElseThrow(() -> new exception);
```

File: src/main/java/mate/jdbc/service/impl/DriverServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import mate.jdbc.dao.DriverDao;
+import mate.jdbc.lib.Inject;
+import mate.jdbc.lib.Service;
+import mate.jdbc.model.Driver;
+import mate.jdbc.service.DriverService;
+
+@Service
+public class DriverServiceImpl implements DriverService {
+    @Inject
+    private DriverDao dao;
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/service/impl/DriverServiceImpl.java:13]:
```suggestion
    private DriverDao driverDao;
```
This service has only one class. But it happens that there can be several.

File: src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+import mate.jdbc.dao.ManufacturerDao;
+import mate.jdbc.lib.Inject;
+import mate.jdbc.lib.Service;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.service.ManufacturerService;
+
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+    @Inject
+    private ManufacturerDao dao;
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java:13]:
```suggestion
    private ManufacturerDao manufacturerDao;
```

File: src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        return dao.create(manufacturer);
+    }
+
+    @Override
+    public Manufacturer get(Long id) {
+        return dao.get(id).get();
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java:22]: Same, as driver service

File: src/main/java/mate/jdbc/Main.java [hunk 11]
```diff
@@ -1,7 +1,55 @@
...
+
 public class Main {
+    private static final Injector injector = Injector.getInstance("mate.jdbc");
+
     public static void main(String[] args) {
-        // test your code here
+        ManufacturerService manufacturerService = (ManufacturerService)
+                injector.getInstance(ManufacturerService.class);
+        Manufacturer manufacturer = new Manufacturer(null, "Alfa Romeo", "Italy");
+        StringBuilder str = new StringBuilder("Info:");
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/Main.java:16]:
```suggestion
        StringBuilder stringBuilder = new StringBuilder("Info:");
```

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**